crates/runtime/src/instance/allocator.rs

```diff
@@ -539,12 +536,21 @@ unsafe fn initialize_vmcontext_globals(instance: &Instance) {
 #[derive(Clone)]
 pub struct OnDemandInstanceAllocator {
     mem_creator: Option<Arc<dyn RuntimeMemoryCreator>>,
+    #[cfg(feature = "async")]
+    stack_size: usize,
 }
 
 impl OnDemandInstanceAllocator {
     /// Creates a new on-demand instance allocator.
-    pub fn new(mem_creator: Option<Arc<dyn RuntimeMemoryCreator>>) -> Self {
-        Self { mem_creator }
+    pub fn new(
+        mem_creator: Option<Arc<dyn RuntimeMemoryCreator>>,
+        #[cfg(feature = "async")] stack_size: usize,
```

Comment:
FWIW I think it's fine to have an extra field like this that isn't conditional, it does add a fair amount of boilerplate to use `#[cfg]` and we're not necessarily getting a whole lot out of it.

Reply:
I can add it back in as I might have gotten carried away with cutting out anything having to do with "execution stacks" from the runtime without `async`.  It definitely added a lot of `cfg` to get right and it clutters up the code.